### ISSUE TITLE
How to limit the number of logs?

### ISSUE THREAD

**Issue by codebuks** [aviggiano/rm-logger#1]
It's a wonderful ideal to store logs in Redis, I  want to do the same thing too.
But something bothering me，how can I limit the number of logs i put into Redis？
How to extends this module ，If I want to store the logs into files when the logs are deleteing form Redis when they are out of limit of number.

**Comment by aviggiano** [comment]
Hey @mbigger, thank you for checking this out.
I created this repository only as a proof of concept to mimic some the abilities of Zookeeper. 
I wanted to create a pubsub server with client discovery. But it's not working as intended.

As for what you said, I believe you can achieve what you want without any redis modules. If you keep a `LIST` with `N+1` fixed elements, that's exactly a limited log store. For that, after each `LPUSH` command, you'd have to issue an `LTRIM 0 N`, to limit the number of items. 

For example:

```
127.0.0.1:6379> LPUSH key val1
(integer) 1
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val1"
127.0.0.1:6379> LPUSH key val2
(integer) 2
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val2"
2) "val1"
127.0.0.1:6379> LPUSH key val3
(integer) 3
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val3"
2) "val2"
3) "val1"
127.0.0.1:6379> LPUSH key val4
(integer) 4
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val4"
2) "val3"
3) "val2"
4) "val1"
127.0.0.1:6379> LPUSH key val5
(integer) 5
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val5"
2) "val4"
3) "val3"
4) "val2"
5) "val1"
127.0.0.1:6379> LPUSH key val6
(integer) 6
127.0.0.1:6379> LTRIM key 0 4
OK
127.0.0.1:6379> LRANGE key 0 -1
1) "val6"
2) "val5"
3) "val4"
4) "val3"
5) "val2"
127.0.0.1:6379> 
```

The `LRANGE` command is only that you see the list shrinking after 5 elements

**Comment by codebuks** [comment]
Thanks for your reply!
The `LRANGE`  command may may not work for me.When i use  `LTRIM ` command to limit the length of logs, the order will lost, i do not know how many logs ware removed, the clients also do not known how to fetch the logs following the previous. One solution is to  implement an  increasing sequence by myself, but that will be hard to maintain. 
When i see the new feature `STREAM` in future version of Redis, i know that is what i want. Every item in a stream has a ID related to time, you can easy to query and limit the length.

**Comment by aviggiano** [comment]
You're right, if you need the order of the logs then this solution won't work.
I read the specification of the `STREAM` data structure and it does seem to suit your needs.

Anyway, have you considered using another log-based software, such as kafka?

**Comment by codebuks** [comment]
@aviggiano ,No, `Redis `+ `Postgres`(MySQL or SQLite) is enough.
